lighthouse-core/computed/metrics/lantern-largest-contentful-paint.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const makeComputedArtifact = require('../computed-artifact.js');
+const LanternMetric = require('./lantern-metric.js');
+const LHError = require('../../lib/lh-error.js');
+const LanternFirstContentfulPaint = require('./lantern-first-contentful-paint.js');
+
+/** @typedef {import('../../lib/dependency-graph/base-node.js').Node} Node */
+
+class LanternLargestContentfulPaint extends LanternMetric {
+  /**
+   * @return {LH.Gatherer.Simulation.MetricCoefficients}
+   */
+  static get COEFFICIENTS() {
+    return {
+      intercept: 0,
+      optimistic: 0.5,
+      pessimistic: 0.5,
+    };
+  }
+
+  /**
+   * TODO: Validate.
+   * @param {Node} dependencyGraph
+   * @param {LH.Artifacts.TraceOfTab} traceOfTab
+   * @return {Node}
+   */
+  static getOptimisticGraph(dependencyGraph, traceOfTab) {
+    const lcp = traceOfTab.timestamps.largestContentfulPaint;
```

Comment:
so right now the impl is identical to FMP except we use the observed LCP for our moment of interest/cutoff? maybe a quick note of this in a fileoverview or for the methods here?

if LCP is as similar to FCP as FMP is this sounds great! if LCP is as different from FCP as I think it is then we'll probably need to get fancier. that'll all shake out in validation though :)

Reply:
I would *very* surprised if we're able to continue with this approach.

> there's not much benefit to landing a placeholder implementation (which needs tests, etc) instead of just returning the observed metric (or whatever) for now to keep the PR simpler.

That sounds like a reasonable way forward to me, though I'm not sure *how* much simpler it really gets given how light this part is :)